keps/sig-apps/2214-indexed-job/README.md

```diff
@@ -94,18 +97,45 @@ own APIs and controllers or adopt third party implementations. Each
 implementation splits the ecosystem, making it harder for higher level systems
 for Job queueing or workflows to support all of them.
 
+Additionally, the Pods within a Job can't easily address and communicate with
+each other, making it hard to run tightly coupled parallel Jobs using the Job
+API.
+
+Third-party operators cover these use cases by defining their own APIs, leading
+to fragmentation of the ecosystem. The operators use mainly two networking
+patterns: (1) fronting each index with a Service or (2) creating Pods with
+stable hostnames based on their index.
+
+Using a Service per index has scalability problems. Other than the Service
+objects themselves, the control plane creates an Endpoint object.
+
+Creating Pods with stable hostnames mitigates this problem. The control plane
+requires only one Service and one Endpoint (or a few EndpointSlices) to inform
+the DNS programming. Pods can address each other with a DNS lookup and
+communicate directly using Pod IPs.
```

Comment:
I really like this argument, it wasn't here last time I read.

Reply:
it was not as clear before :)